python/pyarrow/_dataset.pyx

```diff
@@ -1070,6 +1070,7 @@ cdef class ParquetFileFormat(FileFormat):
         options = &(wrapped.get().reader_options)
         options.use_buffered_stream = read_options.use_buffered_stream
         options.buffer_size = read_options.buffer_size
+        options.enable_parallel_column_conversion = True
```

Comment:
Should the default be False here? (and then be set to True if we only have a single file)